### DISCUSSION TITLE
How to use pretrained weigths

### DISCUSSION THREAD

**Discussion by Michelotti** [MIC-DKFZ/nnUNet#1880] [Q&A]
Hello everyone,

I tried to look at the dedicated section to load the pretrained weigths and train them with another dataset (Pretraining with nnUNet), but I am having trouble to using -pretrained_weigths. 

nnUNetv2_train TARGET_DATASET CONFIG FOLD -pretrained_weights PATH_TO_CHECKPOINT

By following the instruction above, I should insert the correct path, so for example I tried to use as argument for PATH_TO_CHECKPOINT either 

/home/nnUNetFrame/dataset/Dataset114/nnUNet_results/Dataset114/nnUNetTrainer__nnUNetPlans_2d/fold_0/checkpoint_final.pth

or 

/home/nnUNetFrame/dataset/Dataset114/nnUNet_results/Dataset114/nnUNetTrainer__nnUNetPlans_2d/fold_0/checkpoint_final/data.pkl 

Do I make any obvious mistake to load the weigths? Thanks in advance!

Best,
Filippo

**Comment by mariakawula** [comment]
Hi, I might have run into the same problem. I get 
`AttributeError: 'NoneType' object has no attribute 'state_dict'`
Did you manage to solve the problem and perform successful fine-tuning of a pre-trained model?

**Reply by tongchen-z** [reply]
Hi, I met with this problem too. Do you know how to fix it now? Thanks!

**Comment by Edoar-do** [comment]
Hi everyone, 
I think the error is that the nnunet_trainer is not initialize. In fact, if the trainer is not initialized, its network attribute (i.e., the model) is None.
The problem is likely due to the fact that one, included myself, installed nnunetv2 via pip and not by cloning the repo. In fact, when I installed nnunetv2 via pip and ran into this error,  I noticed that my underlying code lacked a significant part in `run_traininig.py` that, instead, is present in the repository's code:

```
if not nnunet_trainer.was_initialized: # this if statement was missing in my code installed via pip
                nnunet_trainer.initialize() 
load_pretrained_weights(nnunet_trainer.network, pretrained_weights_file, verbose=True)
```

I suggest you to check that `if` existence and to add it yourself or to clone the repo as other things can still be missing in the version installed with pip
